test: more accurate checking of commit title

mitake opened the pull request:
I think the correct grammar of commit title would be:
```
<package>{", "<package>}: <description>
```

/cc @heyitsanthony

xiang90 commented:
I will let @heyitsanthony review this. You need to rebase the auth checking on after we get this merged.

heyitsanthony commented on test:
```diff
@@ -158,9 +158,18 @@ function fmt_tests {
 			continue
 		fi
-		if ! [[ "$pfx" =~ :$ ]]; then
+
```
should probably just replace `pfx` with `commitMsg=```echo "$l" | cut -f2- -d' '`

heyitsanthony replied:
although that breaks the current merge/revert checking; I think that would then instead be `if [[ "$commitMsg" == Revert* ]]`

heyitsanthony commented on test:
```diff
@@ -158,9 +158,18 @@ function fmt_tests {
 		fi
-		if ! [[ "$pfx" =~ :$ ]]; then
+
+		pkgPrefix=`echo $l | awk -F ':' '{print $1}'`
```
pkgPrefix=`echo "$commitMsg" | cut -f1 -d':'`

heyitsanthony commented on test:
```diff
@@ -158,9 +158,18 @@ function fmt_tests {
-		if ! [[ "$pfx" =~ :$ ]]; then
+
+		pkgPrefix=`echo $l | awk -F ':' '{print $1}'`
+		if [ "$l" == "$pkgPrefix" ]; then
```
can avoid the duplicate error messages and be stricter about commas:

```
spaceCommas=`echo "$commitMsg" | sed 's/ /\n/g' | grep -c ',$' || echo 0`
commaSpaces=`echo "$commitMsg" | sed 's/,/\n/g' | grep -c '^ ' || echo 0`
if [ "$commitMsg" == "$pkgPrefix" ] || [ "$spaceCommas" != "$commaSpaces" ]]; then
```

heyitsanthony commented on test:
```diff
@@ -158,9 +158,18 @@ function fmt_tests {
+			echo "Got: $l"
+			exit 255
+		fi
+		if [[ "$pkgPrefix" =~ ' '$ ]]; then
```
not needed with above code

heyitsanthony commented:
sorry for botching the grammar, thanks for looking into this

mitake commented:
@heyitsanthony thanks for your refinements. Updated for your comments, PTAL.

I also added a new condition for checking a commit title has at least one : and more than one characters: ``echo $commitMsg | grep -c ":..*"` == 0`

heyitsanthony commented on test:
```diff
@@ -149,18 +149,22 @@ function fmt_tests {
-		pfx=`echo "$l" | cut -f2 -d' '`
-		if [ "$pfx" == "Merge" ]; then
+		commitMsg=`echo "$l" | cut -f2- -d' '`
+		if [[ "$commitMsg" =~ Merge* ]]; then
```
`==` instead of `=~` -- the `=~` will match any substring, `==` matches just the beginning

mitake replied:
Sorry for that, I misunderstood your intention. I'll fix it soon.

heyitsanthony commented on test:
```diff
@@ -149,18 +149,22 @@ function fmt_tests {
 			continue
 		fi
-		if [ "$pfx" == "Revert" ]; then
+		if [[ "$commitMsg" =~ Revert* ]]; then
```
same here

heyitsanthony commented:
lgtm after the `==` fixups

mitake commented:
@heyitsanthony updated for the `==`, PTAL

heyitsanthony commented:
lgtm

mitake commented:
@heyitsanthony thanks for your review and refinements.

@xiang90 could you review? Thanks in advance!

xiang90 commented:
lgtm

mitake commented:
@xiang90 thanks, can I merge this?

xiang90 commented:
@mitake yes. just merge it.

mitake commented:
@xiang90 merged, thanks!
